### PR TITLE
feat: airship-exploration-result reporter

### PR DESCRIPTION
**Pull request by shambarick**

### CODE REVIEW

File: apps/client/src/app/core/data-reporting/exploration-result.reporter.ts [hunk 1]
```diff
@@ -1,5 +1,62 @@
...
+import { DataReporter } from './data-reporter';
+import { Observable } from 'rxjs/Observable';
 
-export interface ExplorationResultReporter {
-  getExplorationType(): ExplorationType;
+export abstract class ExplorationResultReporter implements DataReporter{
+  abstract getDataReports(packets$: Observable<any>): Observable<any[]>;
+  abstract getExplorationType(): ExplorationType;
+
+  createReportsList(stats, resultLog) {
```
Comment (Supamiu) [inline apps/client/src/app/core/data-reporting/exploration-result.reporter.ts:9]: Could be useful to type the output of this method for better maintainability 😄

File: apps/data-extraction/src/extractors/airship-parts.extractor.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          columns: 'ID,Slot,Rank,Components,Surveillance,Retrieval,Speed,Range,Favor,RepairMaterials'
+        }))
+          .pipe(
+            map((page) => page.Results),
+            tap((partResults) => {
+              partResults.forEach((part) => {
+                parts[part.ID] = {
+                  id: part.ID,
+                  // TODO: Waiting for XIVAPI update for the Slot field
+                  slot: Math.trunc(+part.ID / 7), // part.Slot,
```
Comment (Supamiu) [inline apps/data-extraction/src/extractors/airship-parts.extractor.ts:33]: This part will be done next tuesday, you can already implement it as if it was here 👍

**Review by Supamiu — CHANGES_REQUESTED** [review]

File: apps/client/src/app/core/data-reporting/exploration-result.reporter.ts (hunk 1 above)
Comment (shambarick) [reply apps/client/src/app/core/data-reporting/exploration-result.reporter.ts:9]: Sure, where should the model for it be? In the same folder?

Comment (Supamiu) [reply apps/client/src/app/core/data-reporting/exploration-result.reporter.ts:9]: Yeah, same folder is perfect.

**Review by Supamiu — APPROVED** [review]

**Pull request merged**